server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4477,6 +4476,11 @@ private UserVmVO commitUserVm(final boolean isImport, final DataCenter zone, fin
             }
         }
 
+        if (customParameters.containsKey(VmDetailConstants.KVM_VNC_PASSWORD)
+                && StringUtils.isNotEmpty(customParameters.get(VmDetailConstants.KVM_VNC_PASSWORD))) {
+            vm.setVncPassword(customParameters.get(VmDetailConstants.KVM_VNC_PASSWORD));
+        }
```

Comment:
can this be extracted to a separate method.
Is this relevant to the enhancement or a generic fix found on the way?

Reply:
This is relevant to the enhancement as it is used for displaying vnc console for kvm. I've made changes as per your suggestion.